src/EditorFeatures/Core.Wpf/InlineRename/UI/Adornment/InlineRenameAdornment.xaml

```diff
@@ -37,30 +41,83 @@
-                          IsEnabled="{Binding Path=AllowFileRename, Mode=OneWay}" 
-                          Visibility="{Binding Path=ShowFileRename, Converter={StaticResource BooleanToVisibilityConverter}, Mode=OneWay}"/>
+                                <DataTrigger Binding="{Binding IsCollapsed}" Value="False">
+                                    <Setter Property="Moniker" Value="{x:Static imagecatalog:KnownMonikers.CollapseUp}" />
+                                </DataTrigger>
+                            </Style.Triggers>
+                        </Style>
+                    </imaging:CrispImage.Style>
+                </imaging:CrispImage>
+            </Button>
+            <TextBox 
+                Grid.Column="1"
+                x:Name="IdentifierTextBox" 
+                Text="{Binding IdentifierText, UpdateSourceTrigger=PropertyChanged}" 
+                GotFocus="IdentifierTextBox_GotFocus" 
+                HorizontalAlignment="Stretch" />
+        </Grid>
 
-                    <CheckBox Name="PreviewChangesCheckbox" Content="{Binding ElementName=control, Path=PreviewChanges}" Margin="0,8,0,0" IsChecked="{Binding Path=DefaultPreviewChangesFlag, Mode=TwoWay}" />
-                </StackPanel>
-            </Expander.Content>
-        </Expander>
+        <!-- Expanded View -->
+        <StackPanel Orientation="Vertical" Visibility="{Binding IsExpanded, Converter={StaticResource BooleanToVisibilityConverter}}">
+            <CheckBox Content="{Binding ElementName=control, Path=RenameOverloads}" Margin="0,5,0,0" IsChecked="{Binding Path=RenameOverloadFlag, Mode=TwoWay}"
+                      Name="OverloadsCheckbox" Visibility="{Binding RenameOverloadsVisibility}" IsEnabled="{Binding IsRenameOverloadsEditable}" />
+            <CheckBox Name="CommentsCheckbox" Content="{Binding ElementName=control, Path=SearchInComments}" Margin="0,5,0,0" IsChecked="{Binding Path=RenameInCommentsFlag, Mode=TwoWay}" />
+            <CheckBox Name="StringsCheckbox" Content="{Binding ElementName=control, Path=SearchInStrings}" Margin="0,5,0,0" IsChecked="{Binding Path=RenameInStringsFlag, Mode=TwoWay}" />
+            <CheckBox Name="FileRenameCheckbox" 
+                    Content="{Binding Path=FileRenameString}" 
+                    Margin="0,5,0,0"
+                    IsChecked="{Binding Path=RenameFileFlag, Mode=TwoWay}" 
+                    IsEnabled="{Binding Path=AllowFileRename, Mode=OneWay}" 
+                    Visibility="{Binding Path=ShowFileRename, Converter={StaticResource BooleanToVisibilityConverter}, Mode=OneWay}"/>
+        </StackPanel>
+
+        <!-- Collapsed View -->
+        <StackPanel Orientation="Horizontal" Visibility="{Binding IsCollapsed, Converter={StaticResource BooleanToVisibilityConverter}}">
+            <local:InlineRenameOptionIcon Moniker="{x:Static imagecatalog:KnownMonikers.OverloadBehavior}" IsSelected="{Binding Path=RenameOverloadFlag, Mode=TwoWay}"/>
```

Comment:
Can these reuse the checkbox contents as tooltips?

Reply:
Added. Right now they still have accelerator keys in the string though :/ 

Will fix in a follow up. 

![image](https://user-images.githubusercontent.com/475144/144323351-f424c611-c20d-4611-8bd6-a06fdd4e9d8c.png)